fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/rescheduleloan/data/LoanRescheduleRequestDataValidatorImpl.java

```diff
@@ -0,0 +1,382 @@
+import org.apache.fineract.infrastructure.core.data.DataValidatorBuilder;
+import org.apache.fineract.infrastructure.core.exception.InvalidJsonException;
+import org.apache.fineract.infrastructure.core.exception.PlatformApiDataValidationException;
+import org.apache.fineract.infrastructure.core.serialization.FromJsonHelper;
+import org.apache.fineract.infrastructure.core.service.DateUtils;
+import org.apache.fineract.portfolio.loanaccount.domain.Loan;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanCharge;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanRepaymentScheduleInstallment;
+import org.apache.fineract.portfolio.loanaccount.loanschedule.domain.LoanScheduleType;
+import org.apache.fineract.portfolio.loanaccount.rescheduleloan.RescheduleLoansApiConstants;
+import org.apache.fineract.portfolio.loanaccount.rescheduleloan.domain.LoanRescheduleRequest;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+@Component("loanRescheduleRequestDataValidator")
+@AllArgsConstructor
+public class LoanRescheduleRequestDataValidatorImpl implements LoanRescheduleRequestDataValidator {
+
+    public static final Set<String> CREATE_REQUEST_DATA_PARAMETERS = new HashSet<>(
+            Arrays.asList(RescheduleLoansApiConstants.localeParamName, RescheduleLoansApiConstants.dateFormatParamName,
+                    RescheduleLoansApiConstants.graceOnPrincipalParamName,
+                    RescheduleLoansApiConstants.recurringMoratoriumOnPrincipalPeriodsParamName,
+                    RescheduleLoansApiConstants.graceOnInterestParamName, RescheduleLoansApiConstants.extraTermsParamName,
+                    RescheduleLoansApiConstants.rescheduleFromDateParamName, RescheduleLoansApiConstants.newInterestRateParamName,
+                    RescheduleLoansApiConstants.rescheduleReasonIdParamName, RescheduleLoansApiConstants.rescheduleReasonCommentParamName,
+                    RescheduleLoansApiConstants.submittedOnDateParamName, RescheduleLoansApiConstants.loanIdParamName,
+                    RescheduleLoansApiConstants.adjustedDueDateParamName, RescheduleLoansApiConstants.recalculateInterestParamName,
+                    RescheduleLoansApiConstants.endDateParamName, RescheduleLoansApiConstants.emiParamName));
+    private static final Set<String> REJECT_REQUEST_DATA_PARAMETERS = new HashSet<>(
+            Arrays.asList(RescheduleLoansApiConstants.localeParamName, RescheduleLoansApiConstants.dateFormatParamName,
+                    RescheduleLoansApiConstants.rejectedOnDateParam));
+    public static final Set<String> APPROVE_REQUEST_DATA_PARAMETERS = new HashSet<>(
+            Arrays.asList(RescheduleLoansApiConstants.localeParamName, RescheduleLoansApiConstants.dateFormatParamName,
+                    RescheduleLoansApiConstants.approvedOnDateParam));
+    private final FromJsonHelper fromJsonHelper;
+    @Qualifier("progressiveLoanRescheduleRequestDataValidatorImpl")
```

Comment:
in this case to separate Cumulative and Progressive loan validation logic thorugh an abstract class maybe more clean

Reply:
I was thinking on it, but to be honest there is not separate cumulative and progressive loan, it is cumulative is the legacy and de facto implementation and progressive is just sit top of it and override the things that shall behave differently.

Reply:
okay, let's do this way